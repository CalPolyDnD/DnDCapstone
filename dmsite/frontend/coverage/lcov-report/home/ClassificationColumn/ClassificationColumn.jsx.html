<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for home/ClassificationColumn/ClassificationColumn.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">home/ClassificationColumn</a> ClassificationColumn.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/9</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import 'filepond/dist/filepond.min.css';
import {
  Input, Card, Button, CardBody, ListGroup, ListGroupItem, CardHeader,
} from 'reactstrap';
import PropTypes from 'prop-types';
import { FileObject } from '../../Model/FileObject';
&nbsp;
class ClassificationColumn extends Component {
<span class="fstat-no" title="function not covered" >  _h</span>andleClick(e) {
<span class="cstat-no" title="statement not covered" >    e.preventDefault();</span>
  }
&nbsp;
  // TODO: fix this
  isSensitive(val) {
    <span class="missing-if-branch" title="else path not taken" >E</span>if (val === "1") {
        return "Sensitive!";
     }
<span class="cstat-no" title="statement not covered" >    return "";</span>
  }
&nbsp;
  renderClassificationCell() {
    const { file } = this.props;
&nbsp;
    return file.classifications.map(<span class="fstat-no" title="function not covered" >(o</span>bj, index) =&gt; {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;ListGroupItem key={ index } style={{ backgroundColor: '#3d3d3d', color: 'white' }}&gt;
            &lt;div&gt;
              &lt;p&gt;{obj.name}&lt;/p&gt;
              &lt;p style={{ color: 'red', right: "10%"}}&gt;{this.isSensitive(obj.is_sensitive)}&lt;/p&gt;
            &lt;/div&gt;
        &lt;/ListGroupItem&gt;
      );
    });
  };
&nbsp;
  render() {
    return (
      &lt;Card style={{ borderWidth: 0 }} id="classifications"&gt;
        &lt;CardHeader tag="h3" style={{ backgroundColor: '#303030', color: 'white' }}&gt;Classifications&lt;/CardHeader&gt;
        &lt;CardBody style={{ backgroundColor: '#3d3d3d', color: 'white' }}&gt;
          &lt;ListGroup flush&gt;
            {this.renderClassificationCell()}
          &lt;/ListGroup&gt;
          &lt;Input
            placeholder="Search Classification"
            className="mt-3"
            style={{ backgroundColor: '#303030', borderWidth: 0 }}
          /&gt;
          &lt;Button
            onClick={this.handleClick}
            color="primary"
            className="mr-0 btn-block mt-2 mb-2"
          &gt;
            Edit Classifications
          &lt;/Button&gt;
        &lt;/CardBody&gt;
      &lt;/Card&gt;
    );
  }
}
&nbsp;
ClassificationColumn.propTypes = {
  file: PropTypes.instanceOf(FileObject),
};
&nbsp;
export default ClassificationColumn;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 30 2019 00:03:06 GMT-0700 (PDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
