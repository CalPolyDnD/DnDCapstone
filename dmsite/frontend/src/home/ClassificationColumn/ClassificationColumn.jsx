import React, { Component } from 'react';
import 'filepond/dist/filepond.min.css';
import {
    Input, Card, Button, CardBody, ListGroup, ListGroupItem, CardHeader
} from 'reactstrap';
import PropTypes from 'prop-types';
import { FileObject } from '../../Model/FileObject';

class ClassificationColumn extends Component {
  _handleClick(e) {
    e.preventDefault();
  }

  // TODO: at some point, base this off of the classification lookup, not on the file's classifications
  isSensitive(val) {
    if (val === true) { // == works, but === doesn't, for bools or numbers - need to fix
        return "Sensitive!";
     }
    return "";
  }

  renderClassificationCell() {
    const { file } = this.props;

    return file.classifications.map((obj, index) => {
      return (
        <ListGroupItem key={ index } style={{ backgroundColor: '#3d3d3d', color: 'white' }}>
            <div>
              <p>{obj.name}</p>
              <p style={{ color: 'red',     right: "10%"}}>{this.isSensitive(obj.is_sensitive)}</p>
            </div>
        </ListGroupItem>
      );
    });
  };

  render() {
    return (
      <Card style={{ borderWidth: 0 }} id="classifications">
        <CardHeader tag="h3" style={{ backgroundColor: '#303030', color: 'white' }}>Classifications</CardHeader>
        <CardBody style={{ backgroundColor: '#3d3d3d', color: 'white' }}>
          <ListGroup flush>
            {this.renderClassificationCell()}
          </ListGroup>
          <Input
            placeholder="Search Classification"
            className="mt-3"
            style={{ backgroundColor: '#303030', borderWidth: 0 }}
          />
          <Button
            onClick={this.handleClick}
            color="primary"
            className="mr-0 btn-block mt-2 mb-2"
          >
            Edit Classifications
          </Button>
        </CardBody>
      </Card>
    );
  }
}

ClassificationColumn.propTypes = {
  file: PropTypes.instanceOf(FileObject),
};

export default ClassificationColumn;
