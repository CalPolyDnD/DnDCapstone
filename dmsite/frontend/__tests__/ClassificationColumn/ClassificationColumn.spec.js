import React from 'react';
import renderer from 'react-test-renderer';
import ClassificationColumn from '../../src/home/ClassificationColumn/ClassificationColumn';

it('renders correctly with no classifications', () => {
  const file = {
    name: 'testFile',
    path: 'path',
    classifications: [],
    header: 'test header',
  };
  const component = renderer
    .create(<ClassificationColumn file={file} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it('renders correctly', () => {
  const file = {
    name: 'testFile',
    path: 'path',
    classifications: [],
    header: 'test header',
  };
  const component = renderer
    .create(<ClassificationColumn file={file} />);
  const instance = component.getInstance();
  expect(instance.isSensitive('1')).toBe('Sensitive!');
});
