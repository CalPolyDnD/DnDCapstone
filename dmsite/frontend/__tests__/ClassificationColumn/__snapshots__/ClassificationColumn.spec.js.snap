// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no classifications 1`] = `
<div
  className="card"
  id="classifications"
  style={
    Object {
      "borderWidth": 0,
    }
  }
>
  <h3
    className="card-header"
    style={
      Object {
        "backgroundColor": "#303030",
        "color": "white",
      }
    }
  >
    Classifications
  </h3>
  <div
    className="card-body"
    style={
      Object {
        "backgroundColor": "#3d3d3d",
        "color": "white",
      }
    }
  >
    <ul
      className="list-group list-group-flush"
    />
    <input
      className="mt-3 form-control"
      placeholder="Search Classification"
      style={
        Object {
          "backgroundColor": "#303030",
          "borderWidth": 0,
        }
      }
      type="text"
    />
    <button
      aria-label={null}
      className="mr-0 btn-block mt-2 mb-2 btn btn-primary"
      onClick={[Function]}
    >
      Edit Classifications
    </button>
  </div>
</div>
`;
